sdk/go/common/resource/plugin/provider_plugin.go

```diff
@@ -2068,6 +2088,10 @@ func (p *provider) GetPluginInfo(ctx context.Context) (workspace.PluginInfo, err
 		version = &sv
 	}
 
+	if p.overrideVersion != nil {
```

Comment:
If we have an overrideVersion you don't even need to call GetPluginInfo over grpc

Reply:
Makes sense, changed.